GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -474,6 +477,51 @@ protected override void OnLoad(EventArgs e)
 
             // All app init is done, make all repo related similar to switching repos
             SetGitModule(this, new GitModuleEventArgs(new GitModule(Module.WorkingDir)));
+            bool isDashboard = _dashboard?.Visible ?? false;
+            ThreadHelper.FileAndForget(Task.Run(async () =>
+            {
+                if (isDashboard)
+                {
+                    PluginRegistry.InitializeGitHostersOnly();
```

Comment:
Have you noticed that `SetGitModule` synchronously calls `RegisterPlugins`?

Reply:
I think the plugins have already been loaded by the `SetGitModule` call right before this `FileAndForget`.

Reply:
In `SetGitModule` , only the the registration in the UI is done using the content of `PluginRegistry.Plugins` but the filling of this collection is not done in `SetGitModule` so until it is filled it's like if nothing is done...
And so, here we are loading asynchronously to not block the UI (i.e. loading of revision grid or dashboard)